type ‘struct _debug_report_data’ violates the C++ One Definition Rule

**Environment:**
 - OS: Ubuntu 23.04
 - GPU: N/A
 - SDK or header version if building from repo: main (as of 2023-06-21)
 - Options enabled: LTO

**Describe the Issue**

Modified the build by adding `set(CMAKE_INTERPROCEDURAL_OPTIMIZATION ON)` to top level CMakeLists, then build in release mode.

`vvl::unordered_map<uint64_t, std::string> debugObjectNameMap;` is getting ODR violations, likely due to not all components of the build correctly setting the `USE_ROBIN_HOOD_HASHING` build option.  I did not modify `USE_ROBIN_HOOD_HASHING` in my build at all for reference


**Expected behavior**

No warnings/errors

**Additional context**
Main compiler output to look at:
```
[build] /home/cdgiessen/lunarg/Vulkan-ValidationLayers/layers/./error_message/logging.h:162:16: note: type name ‘std::unordered_map<unsigned long, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >, std::hash<unsigned long>, std::equal_to<unsigned long>, std::allocator<std::pair<unsigned long const, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > > > >’ should match type name ‘robin_hood::detail::Table<true, 80ul, unsigned long, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >, robin_hood::hash<unsigned long, void>, std::equal_to<unsigned long> >’
[build]       |                ^
```


Full Compiler output:
```
[proc] Executing command: /usr/bin/cmake --build /home/cdgiessen/lunarg/Vulkan-ValidationLayers/build --config Release --target all --
[build] [1/3  33% :: 1223.214] Linking CXX shared module layers/libVkLayer_khronos_validation.so
[build] [3/3  66% :: 1239.116] Linking CXX shared module tests/layers/libVkLayer_device_profile_api.so
[build] /home/cdgiessen/lunarg/Vulkan-ValidationLayers/layers/./error_message/logging.h:162:16: warning: type ‘struct _debug_report_data’ violates the C++ One Definition Rule [-Wodr]
[build]   162 | typedef struct _debug_report_data {
[build]       |                ^
[build] /home/cdgiessen/lunarg/Vulkan-ValidationLayers/layers/./error_message/logging.h:162:16: note: a different type is defined in another translation unit
[build]   162 | typedef struct _debug_report_data {
[build]       |                ^
[build] /home/cdgiessen/lunarg/Vulkan-ValidationLayers/layers/./error_message/logging.h:166:47: note: the first difference of corresponding definitions is field ‘debugObjectNameMap’
[build]   166 |     vvl::unordered_map<uint64_t, std::string> debugObjectNameMap;
[build]       |                                               ^
[build] /home/cdgiessen/lunarg/Vulkan-ValidationLayers/layers/./error_message/logging.h:166:47: note: a field of same name but different type is defined in another translation unit
[build]   166 |     vvl::unordered_map<uint64_t, std::string> debugObjectNameMap;
[build]       |                                               ^
[build] /home/cdgiessen/lunarg/Vulkan-ValidationLayers/layers/./error_message/logging.h:162:16: note: type name ‘std::unordered_map<unsigned long, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >, std::hash<unsigned long>, std::equal_to<unsigned long>, std::allocator<std::pair<unsigned long const, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > > > >’ should match type name ‘robin_hood::detail::Table<true, 80ul, unsigned long, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >, robin_hood::hash<unsigned long, void>, std::equal_to<unsigned long> >’
[build]   162 | typedef struct _debug_report_data {
[build]       |                ^
[build] [3/3 100% :: 1346.246] Linking CXX executable tests/vk_layer_validation_tests
[build] /home/cdgiessen/lunarg/Vulkan-ValidationLayers/layers/./error_message/logging.h:162:16: warning: type ‘struct _debug_report_data’ violates the C++ One Definition Rule [-Wodr]
[build]   162 | typedef struct _debug_report_data {
[build]       |                ^
[build] /home/cdgiessen/lunarg/Vulkan-ValidationLayers/layers/error_message/logging.h:162:16: note: a different type is defined in another translation unit
[build]   162 | typedef struct _debug_report_data {
[build]       |                ^
[build] /home/cdgiessen/lunarg/Vulkan-ValidationLayers/layers/./error_message/logging.h:166:47: note: the first difference of corresponding definitions is field ‘debugObjectNameMap’
[build]   166 |     vvl::unordered_map<uint64_t, std::string> debugObjectNameMap;
[build]       |                                               ^
[build] /home/cdgiessen/lunarg/Vulkan-ValidationLayers/layers/error_message/logging.h:166:47: note: a field of same name but different type is defined in another translation unit
[build]   166 |     vvl::unordered_map<uint64_t, std::string> debugObjectNameMap;
[build]       |                                               ^
[build] /home/cdgiessen/lunarg/Vulkan-ValidationLayers/layers/./error_message/logging.h:162:16: note: type name ‘std::unordered_map<unsigned long, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >, std::hash<unsigned long>, std::equal_to<unsigned long>, std::allocator<std::pair<unsigned long const, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > > > >’ should match type name ‘robin_hood::detail::Table<true, 80ul, unsigned long, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >, robin_hood::hash<unsigned long, void>, std::equal_to<unsigned long> >’
[build]   162 | typedef struct _debug_report_data {
[build]       |                ^
```

--- a/layers/CMakeLists.txt
+++ b/layers/CMakeLists.txt
@@ -78,7 +78,7 @@ target_include_directories(VkLayer_utils PUBLIC . ${API_TYPE})
 find_package(robin_hood CONFIG)
 option(USE_ROBIN_HOOD_HASHING "robin_hood provides faster versions of std::unordered_map and std::unordered_set" ${robin_hood_FOUND})
 if (USE_ROBIN_HOOD_HASHING)
-    target_link_libraries(VkLayer_utils PRIVATE robin_hood::robin_hood)
+    target_link_libraries(VkLayer_utils PUBLIC robin_hood::robin_hood)
     target_compile_definitions(robin_hood::robin_hood INTERFACE USE_ROBIN_HOOD_HASHING)
 endif()
 

